### DISCUSSION TITLE
alternate compilation for some MD files

### DISCUSSION THREAD

**Discussion by ousia** [orbitalquark/textadept#272] [Q&A] [answered]
I have already defined a general method to compile Markdown documents:

``` lua
textadept.run.compile_commands.markdown = function()
  …
end
```

But for some Markdown files, I need and alternate compilation command.

It would be similar to (which doesn’t work for me [only tested on Linux]):

``` lua
keys['ctrl+f9'] = function()
  if LINUX then
    return './"%e.sh" "%e"'
  elseif WIN32 then
    return 'return '.\"%e.bat" "%e"'
  end
end
```

Which would be the right approach for such an alternative compilation method?

Many thanks for your help.

**Comment by orbitalquark** [comment, marked as answer]
I'm not sure I understand. Did putting that "if" conditional inside your `textadept.run.compile_commands.markdown` function not work?

  textadept.run.compile_commands.markdown = function()
    if LINUX then
      return ...
    elseif WIN32 then
      return ...
    end
  end

Using a function that returns compile/run/build/test commands based on context is the right approach.

**Reply by ousia** [reply]
> I'm not sure I understand. Did putting that "if" conditional inside your `textadept.run.compile_commands.markdown` function not work?

Many thanks for your reply, @orbitalquark.

The conditional inside  `textadept.run.compile_commands.markdown` works like a charm.

But I need another compilation routine for some Markdown files. `textadept.run.compile_commands.markdown` is already taken.

> Using a function that returns compile/run/build/test commands based on context is the right approach.

So I guess I may use `textadept.run.test_commands.markdown`.

I thought this would work:

``` lua
keys['ctrl+f9'] = textadept.run.test

textadept.run.test_commands.markdown = function()
  if LINUX then
    return './"%e.sh" "%e"'
  elseif WIN32 then
    return './"%e.bat" "%e"'
  end
end
```

But for some reason, it doesn’t.

**Reply by orbitalquark** [reply]
`textadept.run.test_commands` does not accept language-specific keys. It's for project test commands. You can expand your existing conditional for those "some Markdown files" that need something other than the usual LINUX or WIN32 commands. Something like:

```
  local use_other_routine = false
  textadept.run.compile_commands.markdown = function()
    if use_other_routine then
      return ...
    elseif LINUX then
      return ...
    elseif WIN32 then
      return ...
    end
  end

  keys['f9'] = textadept.run.compile
  keys['ctrl+f9'] = function()
    use_other_routine = true
    textadept.run.compile()
    use_other_routine = false
  end
```

**Reply by ousia** [reply]
This works perfectly fine.
